### PR TITLE
feat: [PL-31292]: give clear warning while proj org delete

### PR DESCRIPTION
**Pull request by swarajcheguri-harness**
### Summary
Added new dialog for deletion of projects and orgs to reconfirm things before customer deletes anything

Test cases are also recorded in below clip

- deleted a project

- deleted a org

#### Screenshots

[screen-capture (44).webm](https://user-images.githubusercontent.com/73115842/218125153-d3b5e511-a754-4a1b-a4d0-7cb216c4e81f.webm)

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/45-projects-orgs/components/OrganizationCard/OrganizationCard.tsx [hunk 1]
```diff
@@ -64,23 +65,35 @@ export const OrganizationCard: React.FC<OrganizationCardProps> = props => {
...
-    intent: Intent.WARNING,
+    intent: Intent.DANGER,
+    customButtons: (
+      <DeleteProjectOrgButtons
+        onCancel={() => {
+          closeDialog()
+        }}
+        onDelete={() => {
+          onDeleteAction()
+        }}
```
Comment (meenakshi-harness) [inline src/modules/45-projects-orgs/components/OrganizationCard/OrganizationCard.tsx:91]: onDelete={onDeleteAction} this should  also work ?

File: src/modules/45-projects-orgs/components/OrganizationCard/OrganizationCard.tsx [hunk 2]
```diff
@@ -64,23 +65,35 @@ export const OrganizationCard: React.FC<OrganizationCardProps> = props => {
...
     titleText: getString('projectsOrgs.confirmDeleteTitle'),
-    confirmButtonText: getString('delete'),
-    cancelButtonText: getString('cancel'),
-    intent: Intent.WARNING,
+    intent: Intent.DANGER,
+    customButtons: (
+      <DeleteProjectOrgButtons
+        onCancel={() => {
+          closeDialog()
+        }}
```
Comment (meenakshi-harness) [inline src/modules/45-projects-orgs/components/OrganizationCard/OrganizationCard.tsx:88]: same

File: src/modules/45-projects-orgs/pages/organizations/__tests__/OrganizationsPage.test.tsx [hunk 3]
```diff
@@ -141,8 +141,12 @@ describe('Org Page List', () => {
       })
       const form = findDialogContainer()
       expect(form).toBeTruthy()
-      const deleteBtn = queryByText(form as HTMLElement, 'delete')
+
       await act(async () => {
+        const chckBox = queryByText(form as HTMLElement, 'projectsOrgs.yesIamSure')
+        fireEvent.click(chckBox!)
+        await waitFor(() => expect(queryByText(form as HTMLElement, 'delete')).not.toBeDisabled())
+        const deleteBtn = queryByText(form as HTMLElement, 'delete')
```
Comment (meenakshi-harness) [inline src/modules/45-projects-orgs/pages/organizations/__tests__/OrganizationsPage.test.tsx:149]: No need of async await on act. 
These two assertions can be outside act 
` await waitFor(() => expect(queryByText(form as HTMLElement, 'delete')).not.toBeDisabled())
  const deleteBtn = queryByText(form as HTMLElement, 'delete')`

File: src/modules/45-projects-orgs/pages/organizations/__tests__/OrganizationsPage.test.tsx [hunk 4]
```diff
@@ -141,8 +141,12 @@ describe('Org Page List', () => {
...
         fireEvent.click(deleteBtn!)
       })
       expect(deleteOrganization).toBeCalled()
```
Comment (meenakshi-harness) [inline src/modules/45-projects-orgs/pages/organizations/__tests__/OrganizationsPage.test.tsx:152]: this should be awaited

**Review by meenakshi-harness — DISMISSED** [review]

File: src/modules/45-projects-orgs/components/OrganizationCard/OrganizationCard.tsx (hunk 2 above)
Comment (swarajcheguri-harness) [reply src/modules/45-projects-orgs/components/OrganizationCard/OrganizationCard.tsx:88]: seems like it cannot be done 
image

**Review by meenakshi-harness — DISMISSED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by meenakshi-harness — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**